lib/rubocop/cop/performance/double_start_end_with.rb

```diff
@@ -0,0 +1,47 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop checks for double `#start_with?` or `#end_with?` calls
+      # separated by `||`. In some cases such calls can be replaced
+      # with an single `#start_with?`/`#end_with?` call.
+      #
+      # @example
+      #
+      #   @bad
+      #   str.start_with?("a") || str.start_with?(Some::CONST)
+      #   str.start_with?("a", "b") || str.start_with?("c")
+      #   var1 = ...
+      #   var2 = ...
+      #   str.end_with?(var1) || str.end_with?(var2)
+      #
+      #   @good
+      #   str.start_with?("a", Some::CONST)
+      #   str.start_with?("a", "b", "c")
+      #   var1 = ...
+      #   var2 = ...
+      #   str.end_with?(var1, var2)
+      class DoubleStartEndWith < Cop
+        MSG = 'Use `str.%{method}(x, ..., y, ...)` ' \
+              'instead of `str.%{method}(x, ...) || str.%{method}(y, ...)`.'
+
+        def on_or(node)
+          method1, args1, method2, args2 = two_start_end_with_calls(node)
+          if args1 &&
+             method1 == method2 &&
+             args1.all?(&:pure?) &&
+             args2.all?(&:pure?)
```

Comment:
Well it was actually you who showed me it :)

On Wed, Jan 6, 2016 at 2:19 PM, Alex Dowad <notifications@github.com> wrote:

> In lib/rubocop/cop/performance/double_start_end_with.rb
> <https://github.com/bbatsov/rubocop/pull/2590#discussion_r48947823>:
>
> > +      #   @good
> > +      #   str.start_with?("a", Some::CONST)
> > +      #   str.start_with?("a", "b", "c")
> > +      #   var1 = ...
> > +      #   var2 = ...
> > +      #   str.end_with?(var1, var2)
> > +      class DoubleStartEndWith < Cop
> > +        MSG = 'Use `str.%{method}(x, ..., y, ...)` ' \
> > +              'instead of `str.%{method}(x, ...) || str.%{method}(y, ...)`.'
> > +
> > +        def on_or(node)
> > +          method1, args1, method2, args2 = two_start_end_with_calls(node)
> > +          if args1 &&
> > +             method1 == method2 &&
> > +             args1.all?(&:pure?) &&
> > +             args2.all?(&:pure?)
>
> When I saw this PR, I wondered if you knew about Node#pure? or not...
> glad to see you found it!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bbatsov/rubocop/pull/2590/files#r48947823>.
>